src/services/logqlMatchers.ts

```diff
@@ -149,12 +149,13 @@ function parseNonPatternFilters(
   // If quoteString is `, we shouldn't need to un-escape anything
   // But if the quoteString is ", we'll need to remove double escape chars, as these values are re-escaped when building the query expression (but not stored in the value/url)
   if (quoteString === '"' && isRegexSelector) {
+    // replace \\ with \
     const replaceDoubleEscape = new RegExp(/\\\\/, 'g');
     lineFilterValue = lineFilterValue.replace(replaceDoubleEscape, '\\');
   } else if (quoteString === '"') {
-    const replaceDoubleQuoteEscape = new RegExp(/\\\\\"/, 'g');
+    // replace \\\" => "
+    const replaceDoubleQuoteEscape = new RegExp(`\\\\\"`, 'g');
```

Comment:
How do you test this, I can't even add multiple \\\\ 😂 ?

Reply:
This is for non-regex:
https://ops.grafana-ops.net/explore?schemaVersion=1&panes=%7B%22kyg%22:%7B%22datasource%22:%22grafanacloud-logs%22,%22queries%22:%5B%7B%22refId%22:%22logs%22,%22expr%22:%22%7Bservice_name%3D%5C%22loki%2Fquerier%5C%22%7D%20%7C%3D%20%60object%5C%5C%5C%5Cr%5C%5C%5C%5Cn%5C%5C%5C%22%5C%22%60%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22grafanacloud-logs%22%7D,%22editorMode%22:%22code%22,%22queryType%22:%22range%22,%22direction%22:%22backward%22%7D%5D,%22range%22:%7B%22from%22:%22now-15m%22,%22to%22:%22now%22%7D,%22panelsState%22:%7B%22logs%22:%7B%22visualisationType%22:%22logs%22%7D%7D%7D%7D&orgId=1

Line filter:
```
|= `object\\r\\n\""`
```

Take it to logs drilldown -> explore -> drilldown -> explore and eventually you'll get a borked-up query

Reply:
Explore
![Screenshot 2025-05-06 at 7 05 18 AM](https://github.com/user-attachments/assets/0b454c47-d97f-4395-813c-a0bed1b7db94)
Logs Drilldown
![Screenshot 2025-05-06 at 7 05 27 AM](https://github.com/user-attachments/assets/d876b7f8-e18b-4e83-a5d9-e53633ea7d3a)
This seems correct, there are no results in explore locally either.